File: app/src/main/java/me/zheteng/android/videosaver/parser/interceptor/VimeoInterceptor.java
package me.zheteng.android.videosaver.parser.interceptor;

import android.support.annotation.NonNull;
import me.zheteng.android.videosaver.parser.RequestInterceptor;
import okhttp3.Request;

/**
 * Vimeo 需要加个Header
 */
public class VimeoInterceptor implements RequestInterceptor {

    private String mId;

    public VimeoInterceptor(String id) {
        mId = id;
    }

    @NonNull
    @Override
    public Request.Builder intercept(Request.Builder builder) {
        return builder
                .header("Content-Type", "application/json")
                .header("Referer", "https://vimeo.com/" + mId);
    }
}

File: app/src/main/kotlin/me/zheteng/android/videosaver/parse/interceptor/VimeoInterceptor.kt
package me.zheteng.android.videosaver.parse.interceptor

import me.zheteng.android.videosaver.parse.RequestInterceptor
import okhttp3.Request

/**
 * Vimeo 需要加个Header
 */
class VimeoInterceptor(private val mId: String) : RequestInterceptor {

    override fun intercept(builder: Request.Builder): Request.Builder {
        return builder
                .header("Content-Type", "application/json")
                .header("Referer", "https://vimeo.com/" + mId)
    }
}
